azurerm/resource_arm_network_security_group.go

```diff
@@ -216,83 +249,174 @@ func resourceArmNetworkSecurityGroupDelete(d *schema.ResourceData, meta interfac
 	return err
 }
 
-func flattenNetworkSecurityRules(rules *[]network.SecurityRule) []interface{} {
-	result := make([]interface{}, 0)
+func expandAzureRmSecurityRules(d *schema.ResourceData) ([]network.SecurityRule, error) {
+	sgRules := d.Get("security_rule").(*schema.Set)
+	rules := make([]network.SecurityRule, 0)
+
+	for _, sgRaw := range sgRules.List() {
+		sgRule := sgRaw.(map[string]interface{})
+
+		if err := validateSecurityRule(sgRule); err != nil {
+			return nil, err
+		}
+
+		name := sgRule["name"].(string)
+		source_port_range := sgRule["source_port_range"].(string)
+		destination_port_range := sgRule["destination_port_range"].(string)
+		source_address_prefix := sgRule["source_address_prefix"].(string)
+		destination_address_prefix := sgRule["destination_address_prefix"].(string)
+		priority := int32(sgRule["priority"].(int))
+		access := sgRule["access"].(string)
+		direction := sgRule["direction"].(string)
+		protocol := sgRule["protocol"].(string)
+
+		properties := network.SecurityRulePropertiesFormat{
+			SourcePortRange:          &source_port_range,
+			DestinationPortRange:     &destination_port_range,
+			SourceAddressPrefix:      &source_address_prefix,
```

Comment:
SourceAddressPrefixes should be expanded too, if not during an update of an existing nsg we will drop enhanced rules.

Reply:
Could you explain what you mean exactly? I'm afraid I don't understand what you mean by `SourceAddressPrefixes should be expanded too`...

Reply:
Maybe you missed that I check and set those fields (if needed) as well? See below starting from line 288...

Reply:
I'm sorry, I made a mistake during my test, your PR fix that problem with #692... And it works as described 👍

Reply:
Check, cool...